include/roboteam_utils/Line.h

```diff
@@ -164,6 +164,14 @@ class Line {
          */
         [[nodiscard]] std::optional<Vector2> intersects(const LineSegment &line) const ;
 
+       /**
+        * @brief Compute the line intersection of this line in positive direction, i.e. in the direction from the start to the end of the given line.
+        *
+        * @param line The line of which the positive direction intersection is computed.
+        * @return No vector if the lines do not intersect in positive direction. Otherwise it returns a vector which represents the position of intersection.
+        */
+        [[nodiscard]] std::optional<Vector2> forwardIntersect(const Line &line) const;
+
```

Comment:
Personally not a fan of this being in this class, I think it should be it's own seperate class. Line objects are meant to represent infinite lines and not halflines being infinite in only one direction. This is even specified in the class description.

Reply:
True, but I am more fan of it being part of the Line class than the LineSegment class